pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -615,7 +618,47 @@ private static Expression toExpression(SqlNode node) {
             funcExpr.getFunctionCall().addToOperands(toExpression(child));
           }
         }
+
+        if (FunctionDefinitionRegistry.isScalarFunc(funcName) && isCompileTimeEvaluationPossible(funcExpr)) {
+
+          ScalarFunctionType scalarFunctionType = ScalarFunctionType.getScalarFunctionType(funcName);
+          switch (scalarFunctionType) {
+            case NOW:
+              funcExpr = RequestUtils.getLiteralExpression(System.currentTimeMillis());
+              break;
+            case FORMAT_DATETIME:
+              //DATETIME_FORMAT ('2020-01-01', 'yyyy-MM-dd')
+              String input = funcExpr.getFunctionCall().getOperands().get(0).getLiteral().getStringValue();
+              String format = funcExpr.getFunctionCall().getOperands().get(1).getLiteral().getStringValue();
+              long output = DateTimeFormat.forPattern(format).parseMillis(input);
+              funcExpr = RequestUtils.getLiteralExpression(output);
+              break;
+            default:
+              //no change, let the expression be handled during execution phase
+          }
+        }
+
         return funcExpr;
     }
   }
+
+  /**
+   * Utility method to check if the function can be evaluated during the query compilation phae
+   * @param funcExpr
+   * @return true if all arguments are literals
+   */
+  private static boolean isCompileTimeEvaluationPossible(Expression funcExpr) {
+
+    boolean compileTimeEvaluationPossible = true;
+    Function functionCall = funcExpr.getFunctionCall();
+    if(functionCall.getOperandsSize() > 0) {
+      for (Expression expression : functionCall.getOperands()) {
+        if (expression.getType() != ExpressionType.LITERAL) {
```

Comment:
this should be recursively evaluated, technically a function of function of literal should still be true here.
E.g. `format_time(now(), 'yyyy-MM-dd')`

Reply:
since logic is depth-first, it should work.
`format_time(now(), 'yyyy-MM-dd')` will first get changed to 
`format_time(123123123123, 'yyyy-MM-dd')` which will be evaluated to
`2020-01-01`